src/info/Info.ts

```diff
@@ -0,0 +1,130 @@
+export class DIVEInfo {
+    private static _supportsWebXR: boolean | null = null;
+
+    /**
+     *
+     * @returns The system the user is using. Possible values are "Android", "iOS", "Windows", "MacOS", "Linux" or "Unknown".
+     */
+    public static GetSystem(): string {
+        const platform = navigator.platform;
+        if (/Android/.test(navigator.userAgent)) {
+            return "Android";
+        } else if (/iPhone|iPad|iPod/.test(navigator.userAgent)) {
+            return "iOS";
+        } else if (platform.startsWith("Win")) {
+            return "Windows";
+        } else if (platform.startsWith("Mac")) {
+            return "MacOS";
+        } else if (platform.startsWith("Linux")) {
+            return "Linux";
+        } else {
+            return "Unknown";
+        }
+    }
+
+    /**
+     * @returns A promise that resolves to a boolean indicating whether the user's device supports WebXR.
+     */
+    public static async GetSupportsWebXR(): Promise<boolean> {
+        return new Promise((resolve) => {
+            if (this._supportsWebXR !== null) {
+                resolve(this._supportsWebXR);
+                return;
+            }
+
+            if (!navigator.xr) {
+                this._supportsWebXR = false;
+                resolve(false);
+                return;
+            }
+            // Check if immersive-vr session mode is supported
+            navigator.xr.isSessionSupported('immersive-ar')
+                .then((supported) => {
+                    if (supported) {
+                        this._supportsWebXR = true;
+                        resolve(true);
+                    } else {
+                        this._supportsWebXR = false;
+                        resolve(false);
+                    }
+                }).catch(() => {
+                    this._supportsWebXR = false;
+                    resolve(false);
+                });
+        });
```

Comment:
This is a async function, why do you wrap it in a Promise?

```suggestion
    public static async GetSupportsWebXR(): Promise<boolean> {
        if (this._supportsWebXR !== null) {
            return this._supportsWebXR;
        }

        if (!navigator.xr) {
            this._supportsWebXR = false;
            return false;
        }
        // Check if immersive-vr session mode is supported
        try {
            const supported = navigator.xr.isSessionSupported('immersive-ar');
            this._supportsWebXR = supported;
            return supported;
        } catch(e) {
            this._supportsWebXR = false;
            return false;
        }
    });
```